packages/stylelint-plugin-mozaic/lib/splitSelector.js

```diff
@@ -70,16 +69,23 @@ function splitBem(string, options) {
       arr.push({ partType: 'block', string: part })
     }
   })
-
   return arr
 }
 
+const removePrefixes = (selector, options) => {
+  const result = options.prefixes
+    ? options.prefixes.filter(prefix => selector.includes(prefix, 0))
+    : []
+  if (result.length === 1) return selector.replace(result[0], '')
+  if (result.length === 0) return selector
+}
+
 const splitSelector = (selector, options) => {
   selector.forEach((part, i) => {
     if (part.type === 'class') {
-      part.bemStructure = splitBem(part.value, options)
+      const value = removePrefixes(part.value, options)
+      part.bemStructure = splitBem(value, options)
```

Comment:
we need to keep the prefix as part of the bem structure, not just remove it.

Reply:
something like this : 

```js
{
    identifier: 'mc-button',
    prefix: 'mc-',
    block: 'button',
    element: 'icon'
    modifier: 'large'
    media: 'from-m'
}
```

we test for `caseStyle` on `block`, `element`, `modifier` and `media`. 

we keep identifier as a kind of ID.

Reply:
```
my-call--small@from-m
[
  { partType: 'block', string: 'my-call' },
  { partType: 'modifier', string: 'small' },
  { partType: 'media', string: 'from-m' },
  { partType: 'prefix', string: 'mc-' }
]
```